rs/tests/driver/src/driver/ic_gateway_vm.rs

```diff
@@ -0,0 +1,385 @@
+const IC_GATEWAY_VM_FILE: &str = "vm.json";
+const IMAGE_PATH: &str = "rs/tests/ic_gateway_uvm_config_image.zst";
+const IC_GATEWAY_VMS_DIR: &str = "ic_gateway_vms";
+const PLAYNET_FILE: &str = "playnet.json";
+const BN_AAAA_RECORDS_CREATED_EVENT_NAME: &str = "bn_aaaa_records_created_event";
+const READY_TIMEOUT: Duration = Duration::from_secs(60);
+const RETRY_INTERVAL: Duration = Duration::from_secs(5);
+
+/// Represents an IC HTTP Gateway VM, it is a wrapper around Farm's Universal VM.
+#[derive(Debug)]
+pub struct IcGatewayVm {
+    universal_vm: UniversalVm,
+}
+
+/// Represents a deployed IC HTTP Gateway VM.
+#[derive(Debug)]
+pub struct DeployedIcGatewayVm {
+    vm: AllocatedVm,
+    https_url: Url,
+}
+
+impl DeployedIcGatewayVm {
+    pub fn get_public_url(&self) -> Url {
+        self.https_url.clone()
+    }
+
+    /// Retrieves the underlying VM.
+    pub fn get_vm(&self) -> AllocatedVm {
+        self.vm.clone()
+    }
+}
+
+impl Default for IcGatewayVm {
+    fn default() -> Self {
+        Self::new(IC_GATEWAY_VM_NAME)
+    }
+}
+
+impl IcGatewayVm {
+    /// Creates a new IC Gateway VM with the specified name.
+    pub fn new(name: &str) -> Self {
+        let universal_vm =
+            UniversalVm::new(name.to_string()).with_config_img(get_dependency_path(IMAGE_PATH));
+        Self { universal_vm }
+    }
+
+    /// Starts the IC Gateway VM, configuring DNS and certificates.
+    pub fn start(&self, env: &TestEnv) -> Result<()> {
+        let logger = env.logger();
+        info!(logger, "Starting IC Gateway VM: {}", self.universal_vm.name);
+
+        // Start the underlying universal VM
+        self.universal_vm.start(env)?;
+        let deployed_vm = env.get_deployed_universal_vm(&self.universal_vm.name)?;
+        let allocated_vm = deployed_vm.get_vm()?;
+
+        // Get IPv6 address and API node URLs
+        let vm_ipv6 = allocated_vm.ipv6.to_string();
+        let api_nodes_urls = self.get_api_nodes_urls(env);
+
+        // Handle playnet configuration and DNS records
+        let playnet = self.load_or_create_playnet(env, &vm_ipv6)?;
+        let bn_fqdn = playnet.playnet_cert.playnet.clone();
+        self.configure_dns_records(env, &playnet, &bn_fqdn)?;
+
+        // Emit log event for AAAA records
+        emit_bn_aaaa_records_event(&logger, &bn_fqdn, playnet.aaaa_records.clone());
+
+        // Save playnet configuration and start the gateway
+        let playnet_url = Url::parse(&format!("https://{}", bn_fqdn))?;
+        env.write_deployed_ic_gateway(&self.universal_vm.name, &playnet_url, &allocated_vm)?;
+        self.start_gateway_container(&deployed_vm, &playnet, api_nodes_urls)?;
+
+        // Wait for the service to become ready.
+        // Readiness is defined when some API boundary node used by ic-gateway responds with HTTP 200 to /api/v2/status.
+        let health_url = playnet_url.join("/api/v2/status")?;
+        let msg = format!("await_status_is_healthy of {}", self.universal_vm.name);
```

Comment:
I would actually log the `health_url` in the `msg`. That makes it easier to debug things in case things break.